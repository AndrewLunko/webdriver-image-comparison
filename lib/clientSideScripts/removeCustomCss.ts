/**
 * Remove the custom style that has been added
 */
export default function removeCustomCss(id: string): void {
  const elem = document.querySelector(`style#${id}`);
  elem.parentNode.removeChild(elem);
}
