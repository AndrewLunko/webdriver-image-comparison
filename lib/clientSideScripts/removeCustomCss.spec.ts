import removeCustomCss from './removeCustomCss';

describe('removeCustomCss', ()=>{
  it('should be able to remove the custom css', ()=>{
    // Set up our document body
    const id = 'test';
    const cssText = 'body:{width:100%}';
    const head = document.head || document.getElementsByTagName('head')[ 0 ];
    const style = document.createElement('style');

    style.id = id;
    style.appendChild(document.createTextNode(cssText));
    head.appendChild(style);

    expect(document.head.textContent).toMatchSnapshot();

    removeCustomCss(id);

    expect(document.head.textContent).toMatchSnapshot();
  });
});
