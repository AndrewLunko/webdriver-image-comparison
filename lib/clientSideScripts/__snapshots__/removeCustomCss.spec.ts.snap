// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeCustomCss should be able to remove the custom css 1`] = `"body:{width:100%}"`;

exports[`removeCustomCss should be able to remove the custom css 2`] = `""`;
